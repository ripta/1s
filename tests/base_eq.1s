-- make it easier to see the result of `eq?`
[ eq? [ "equal" ] [ "inequal" ] if ] [ test ] :

"integers"
1 1 test
1 2 test

"floats"
1.2345 1.2345 test
1.2345 9.8765 test

"strings"
"foo" "foo" test
"foo" "bar" test
"foo" "foo bar" test

"symbols"
#true #true test
#true #false test

"quoted-blocks"
[ ] [ ] test
[ [ ] ] [ ] test
[ ] [ [ ] ] test
[ [ ] ] [ [ ] ] test

"quoted-integers"
[ 1 ] [ 1 ] test
[ 1 ] 1 test
1 [ 1 ] test
[ 1 2 ] [ 1 2 ] test
[ 1 2 ] [ 2 1 ] test
[ 1 2 ] [ 1 2 3 ] test

"quoted-strings"
[ "foo" ] [ "foo" ] test
[ "foo" ] "foo" test
"foo" [ "foo" ] test
[ "foo" ] [ "foo" "bar" ] test
[ "foo" ] [ "foo bar" ] test
[ [ "foo" ] ] [ [ "foo" ] ] test
[ [ "foo" ] ] [ [ "bar" ] ] test
