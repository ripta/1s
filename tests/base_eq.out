Result:
  "integers"
  "equal"
  "inequal"
  "floats"
  "equal"
  "inequal"
  "strings"
  "equal"
  "inequal"
  "inequal"
  "symbols"
  "equal"
  "inequal"
  "quoted-blocks"
  "equal"
  "inequal"
  "inequal"
  "equal"
  "quoted-integers"
  "equal"
  "inequal"
  "inequal"
  "equal"
  "inequal"
  "inequal"
  "quoted-strings"
  "equal"
  "inequal"
  "inequal"
  "inequal"
  "inequal"
  "equal"
  "inequal"
